src/app/public/testing/country-field/country-field-fixture.ts

```diff
@@ -0,0 +1,168 @@
+   */
+  public get autocomplete(): string {
+    const inputEl = this.getInput();
+    return inputEl.nativeElement.getAttribute('autocomplete');
+  }
+
+  /**
+   * A flag indicating if the country flag is currently visible.
+   * The flag will be visible only if a selection has been made
+   * and if the hideSelectedCountryFlag option is false.
+   */
+  public get countryFlagIsVisible(): boolean {
+    const flag = this.getCountryFlag();
+    return flag !== null;
+  }
+
+  /**
+   * A flag indicating whether or not the input has been disabled.
+   */
+  public get isDisabled(): boolean {
+    const inputEl = this.getInput();
+    return inputEl.nativeElement.disabled;
+  }
+
+  /**
+   * The value of the input field.
+   */
+  public get searchText(): string {
+    const inputEl = this.getInput();
+    return inputEl.nativeElement.value;
+  }
+
+  /**
+   * The value of the input field.
+   */
+  public get selectedCountryData(): SkyCountryFieldCountry {
+    return this.countries.find((x: any) => x.name === this.searchText);
+  }
+
+  /**
+   * Enters the search text into the input field displaying search results, but making no selection.
+   * @param searchText The name of the country to select.
+   */
+  public async search(searchText: string): Promise<NodeListOf<HTMLElement>> {
+    return this.searchAndGetResults(searchText, this.fixture);
+  }
+
+  /**
+   * Enters the search text into the input field and selects the first result (if any).
+   * @param searchText The name of the country to select.
+   */
+  public async select(searchText: string): Promise<any> {
+    this.searchAndSelect(searchText, 0, this.fixture);
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  /**
+   * Clears the country selection and input field.
+   */
+  public clear(): Promise<any> {
+    this.enterSearch('', this.fixture);
+
+    this.fixture.detectChanges();
+    return this.fixture.whenStable();
+  }
+
+  //#region helpers
+
+  private getCountryFlag(): DebugElement {
+    return this.debugEl.query(By.css('.sky-country-field-flag'));
+  }
+
+  private getInput(): DebugElement {
+    return this.debugEl.query(By.css('textarea'));
+  }
+
+  private getAutocompleteElement(): HTMLElement {
+    return document.querySelector('.sky-autocomplete-results') as HTMLElement;
+  }
+
+  private getInputElement(): HTMLTextAreaElement {
+    return this.getInput().nativeElement as HTMLTextAreaElement;
+  }
+
+  private blurInput(fixture: ComponentFixture<any>): void {
+    SkyAppTestUtility.fireDomEvent(this.getInputElement(), 'blur');
+    fixture.detectChanges();
+    tick();
+  }
+
+  private enterSearch(newValue: string, fixture: ComponentFixture<any>): void {
+    const inputElement = this.getInputElement();
+    inputElement.value = newValue;
+
+    SkyAppTestUtility.fireDomEvent(inputElement, 'keyup');
+    fixture.detectChanges();
+    tick();
+    fixture.detectChanges();
+    tick();
```

Comment:
Are these `tick()`s doing anything? I thought they only are used when inside Angular's `fakeAsync` zone. We shouldn't need them here. If so, we may need to figure out a better way to do this.